sentinel-core/src/main/java/com/alibaba/csp/sentinel/config/ConfigLoadUtils.java

```diff
@@ -0,0 +1,151 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.csp.sentinel.config;
+
+import com.alibaba.csp.sentinel.util.AppNameUtil;
+import com.alibaba.csp.sentinel.util.StringUtil;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+/**
+ * @author lianglin
+ * @since 2019-05-31 15:19
+ */
+public class ConfigLoadUtils {
+
+    private static String DEFAULT_SENTINEL_CUSTOMIZE_FILE = "sentinel.properties";
+    private static String SENTINEL_CONFIG_FILE = "csp.sentinel.config.file";
+
+    private static final String DIR_NAME = "logs" + File.separator + "csp";
+    private static final String USER_HOME = "user.home";
+
+    private static Properties properties = new Properties();
+
+    static {
+        init();
+    }
+
+    public static Properties loadProperties() {
+        return properties;
+    }
+
+
+    private static void init() {
+
+        //Load from JVM -D user-specified file
+        String fileName = System.getProperty(SENTINEL_CONFIG_FILE);
+        if (StringUtil.isNotBlank(fileName)) {
+            if (fileName.startsWith("/")) {
+                loadPropertiesFromFille(fileName);
+            } else {
+                loadPropertiesFromResource(fileName);
+            }
+        } else {
+            //try load from default file
+            loadPropertiesFromResource(DEFAULT_SENTINEL_CUSTOMIZE_FILE);
+            if (properties.isEmpty()) {
+                //may be old version
+                String userHome = System.getProperty(USER_HOME);
+                String path = addSeparator(userHome) + DIR_NAME + File.separator;
+                AppNameUtil.resolveAppName();
+                String appName = AppNameUtil.getAppName();
+                fileName = path + appName + ".properties";
+                File file = new File(fileName);
+                if (file.exists()) {
+                    loadPropertiesFromFille(fileName);
+                }
+            }
+        }
+        //JVM parameter override file config
+        CopyOnWriteArraySet<Map.Entry<Object, Object>> entries = new CopyOnWriteArraySet<>(System.getProperties().entrySet());
+        for (Map.Entry entry : entries) {
+            properties.setProperty(entry.getKey().toString(), entry.getValue().toString());
```

Comment:
We might also need to log here as before (if the value is replaced).

Reply:
Is better to add a separate log config file?  and the sentinel core config is another file

Reply:
the config load order may be : log config -> sentinel-core config.

Reply:
+1 for this